### ISSUE TITLE
Typescript: smooth brackets around arrow function return value is removed when return value is cast

### ISSUE THREAD

**Issue by frederickfogerty** [prettier/prettier#1861]
Hey there! Both of these inputs produce an incorrect output.

1:
```js
() => ({
  key: 'value'
} as any);
```
2:
```js
() => ({
  key: 'value'
}) as any;
```

Output:
```js
() =>
  {
    key: "value"
  } as any;
```

This does not compile.

Please see: https://prettier.github.io/prettier/#%7B%22content%22%3A%22()%20%3D%3E%20(%7B%5Cn%20%20key%3A%20'value'%5Cn%7D%20as%20any)%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D

and

https://prettier.github.io/prettier/#%7B%22content%22%3A%22()%20%3D%3E%20(%7B%5Cn%20%20key%3A%20'value'%5Cn%7D)%20as%20any%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D

**Comment by azz** [comment]
Thanks for the report! 

This is a duplicate of #1851 and has been fixed in master.

**Comment by vjeux** [comment]
I just published 1.4.1 which fixes it!
